### ISSUE TITLE
Listbox with paging. NullPointerException.

### ISSUE THREAD

**Issue by emorozovs** [tony1223/zkfiddle#187]
Listbox with paging. If dynamically in custom renderer call row.detach(), and then on client select last row, ZK throws java.lang.NullPointerException or java.lang.IndexOutOfBoundsException.
Please see example: http://zkfiddle.org/sample/1esfv0m/41-Another-new-ZK-fiddle#source-3 
Affected till last ZK version.

1)
java.lang.NullPointerException
at org.zkoss.zul.Listbox.setSelectedIndex(Listbox.java:929)
at org.zkoss.zul.Listbox.selectItem(Listbox.java:953)
at org.zkoss.zul.Listbox.service(Listbox.java:3396)
at org.zkoss.zk.ui.impl.DesktopImpl.service(DesktopImpl.java:699)
at org.zkoss.zk.ui.impl.UiEngineImpl.execUpdate(UiEngineImpl.java:1193)
at org.zkoss.zk.au.http.DHtmlUpdateServlet.process(DHtmlUpdateServlet.java:558)
at org.zkoss.zk.au.http.DHtmlUpdateServlet.doGet(DHtmlUpdateServlet.java:456)
at org.zkoss.zk.au.http.DHtmlUpdateServlet.doPost(DHtmlUpdateServlet.java:464)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:595)
at javax.servlet.http.HttpServlet.service(HttpServlet.java:668)
..

2)
java.lang.IndexOutOfBoundsException: Index: -1, Size: 3 
at java.util.ArrayList.get(ArrayList.java:352) 
at org.zkoss.zul.ListModelList.getElementAt(ListModelList.java:150) 
at org.zkoss.zul.Listbox.service(Listbox.java:3423) 
at org.zkoss.zk.ui.impl.DesktopImpl.service(DesktopImpl.java:708) 
at org.zkoss.zk.ui.impl.UiEngineImpl.execUpdate(UiEngineImpl.java:1204) 
at org.zkoss.zk.au.http.DHtmlUpdateServlet.process(DHtmlUpdateServlet.java:601) 
at org.zkoss.zk.au.http.DHtmlUpdateServlet.doGet(DHtmlUpdateServlet.java:483) 
at org.zkoss.zk.au.http.DHtmlUpdateServlet.doPost(DHtmlUpdateServlet.java:492) 
at javax.servlet.http.HttpServlet.service(HttpServlet.java:595) 
at javax.servlet.http.HttpServlet.service(HttpServlet.java:668) 
...

**Comment by emorozovs** [comment]
http://tracker.zkoss.org/browse/ZK-2490
